lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -307,22 +309,40 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
                 redcap_record_instance["pre_scan_barcode"],
                 redcap_record_instance["barcode_swabsend"]]
 
-            specimen_entry, specimen_reference = create_specimen(
-                prioritized_barcodes = prioritized_barcodes,
-                patient_reference = patient_reference,
-                collection_date = get_collection_date(redcap_record_instance, collection_method),
-                sample_received_time = redcap_record_instance['samp_process_date'],
-                able_to_test = redcap_record_instance['able_to_test'],
-                system_identifier = INTERNAL_SYSTEM)
-
-            specimen_observation_entry = create_specimen_observation_entry(
-                specimen_reference = specimen_reference,
-                patient_reference = patient_reference,
-                encounter_reference = initial_encounter_reference)
+            # Check if specimen barcode is valid
+            specimen_barcode = None
+            for barcode in prioritized_barcodes:
+                specimen_barcode = barcode.strip()
+                if specimen_barcode:
+                    # Disable logging when calling find_identifier here to suppess alerts
+                    # for HCT unrecognized barcodes which are being managed in LIMS
+                    logging.disable(logging.WARNING)
+                    specimen_identifier = find_identifier(db, specimen_barcode)
```

Comment:
Ah, clever approach - I think this should work ok! It opens up a small probability of missing other warnings from that method, if any, but since its log output and not a specific exception we could catch, I can see why you took this approach, and I think it will succeed in quieting the warnings.

Some alternative options would be to write the database lookup here directly, without using find_identifier - and then you could leave out warnings, or you could modify find_identifier() to take an optional param to disable the warning, but maybe with a default to warn.

Reply:
I see what you mean, but I'm tempted to leave as is. The `find_identifier` function only has two possible results, the warning we're trying to suppress is the only one that's ever logged if the identifier isn't found:
https://github.com/seattleflu/id3c/blob/1d236e4e4d2b335514721ab2b4ef4861c584b2e0/lib/id3c/db/__init__.py#L94